### ISSUE TITLE
[Bug][Devlake][GitHub] GitHub app connection data source usability issues

### ISSUE THREAD

**Issue by petercort** [apache/devlake#7655]
### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

Creating a GitHub App connection isn't working as expected. 
1. No organizations or repositories are found 'naturally' after the connection is created
2. Can't multiselect to add repositories when they are specified into the datasources search 

![Image](https://github.com/user-attachments/assets/f4a2aa92-38af-48c8-aa4b-b6f34896b864)
![Image](https://github.com/user-attachments/assets/ea3cae2c-6e86-4cc8-9f7e-a914635b58a8)
![Image](https://github.com/user-attachments/assets/6510a14b-8836-4b28-a3f1-804ea0067346)
![Image](https://github.com/user-attachments/assets/a5d3dc25-a083-4ddc-8951-cfe4fe599382)
![Image](https://github.com/user-attachments/assets/207a6333-fa2f-4a7f-bac5-3178ab4e38ae)

### What do you expect to happen

I would expect that the repositories in the organization where the app has been installed would auto populate and I could select multiples to add as a data source.  

### How to reproduce

1. Configure a GitHub App as a connection
2. Add data sources - no 

### Anything else

I was able to reproduce this behavior with a GitHub app installed into an organization and my personal account. 

I'm willing to take a crack at a PR if it's not a massive thing. I'll spend some time setting up a debugger and seeing if I can get more info. Might need someone more skilled than me (cough cough copilot) to fix this. 

I also tested this with the v0.21.0@ac4f7fe and it's reproducible there. 

Here's the error message formatted nicely: 

```
level=error msg="HTTP 400 error
    caused by: attached stack trace
     -- stack trace:
        | github.com/apache/incubator-devlake/helpers/pluginhelper/api.(*DsScopeApiHelper[...]).PutMultiple
        | /app/helpers/pluginhelper/api/ds_scope_api_helper.go:114  
        | github.com/apache/incubator-devlake/plugins/github/api.PutScopes  
        | /app/plugins/github/api/scope_api.go:42
        | github.com/apache/incubator-devlake/server/api.handlePluginCall.func1
        | /app/server/api/router.go:135
        | github.com/gin-gonic/gin.(*Context).Next
        | /go/pkg/mod/github.com/gin-gonic/gin@v1.9.1/context.go:174
        | github.com/apache/incubator-devlake/server/api.OAuth2ProxyAuthentication.func1
        | /app/server/api/middlewares.go:95
        | github.com/gin-gonic/gin.(*Context).Next
        | /go/pkg/mod/github.com/gin-gonic/gin@v1.9.1/context.go:174
        | github.com/apache/incubator-devlake/server/api.RestAuthentication.func1
        | /app/server/api/middlewares.go:117
        | github.com/gin-gonic/gin.(*Context).Next
        | /go/pkg/mod/github.com/gin-gonic/gin@v1.9.1/context.go:174
        | github.com/gin-gonic/gin.CustomRecoveryWithWriter.func1
        | /go/pkg/mod/github.com/gin-gonic/gin@v1.9.1/recovery.go:102
        | github.com/gin-gonic/gin.(*Context).Next
        | /go/pkg/mod/github.com/gin-gonic/gin@v1.9.1/context.go:174
        | github.com/gin-gonic/gin.LoggerWithConfig.func1
        | /go/pkg/mod/github.com/gin-gonic/gin@v1.9.1/logger.go:240
        | github.com/gin-gonic/gin.(*Context).Next
        | /go/pkg/mod/github.com/gin-gonic/gin@v1.9.1/context.go:174
        | github.com/gin-gonic/gin.(*Engine).handleHTTPRequest
        | /go/pkg/mod/github.com/gin-gonic/gin@v1.9.1/gin.go:620
        | github.com/gin-gonic/gin.(*Engine).ServeHTTP
        | /go/pkg/mod/github.com/gin-gonic/gin@v1.9.1/gin.go:576
        | net/http.serverHandler.ServeHTTP
        | /usr/local/go/src/net/http/server.go:2936
        | net/http.(*conn).serve
        | /usr/local/go/src/net/http/server.go:1995
        | runtime.goexit
        | /usr/local/go/src/runtime/asm_arm64.s:1172
    Wraps: (2) invalid data row (400)
    Error types: (1) *withstack.withStack (2) *errutil.leafError"
```

### Version

v1.0.0-beta11@15c4c09

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by klesh** [comment]
Can you share the request payload from your browser under the Network tab? 
Please remove any sensitive information before you post but keep the overall structure.

**Comment by petercort** [comment]
## For the connection being created and scopes loading: 
Once the connection is created, I've got 2 calls 
http://localhost:4000/api/plugins/github/connections/11/scopes?page=1&pageSize=10&blueprints=true
Response:
```
{"count":0,"scopes":[]}
```

http://localhost:4000/api/plugins/github/connections/11/remote-scopes
Response:
  ```
{
    "children": [
        {
            "type": "scope",
            "parentId": "petes-fre-shavocado-stand",
            "id": "UID",
            "name": "roadwork",
            "fullName": "petes-fre-shavocado-stand/roadwork",
            "data": null
        }
    ],
    "nextPageToken": ""
}
```

## Multi selecting scope on a manual search 
http://localhost:4000/api/plugins/github/connections/11/scopes

![payload for adding multiple repos](https://github.com/user-attachments/assets/4f04aefb-3ac2-46c7-a39f-d7bc8160d074)

**Comment by Mantesh-Globant** [comment]
@petercort We are facing same issue while integrating Github app with devlake , we have validated all the required permissions and end point url and still facing same issue
@klesh

**Comment by klesh** [comment]
Seems like there are multiple problems to me:
1. http://localhost:4000/api/plugins/github/connections/11/remote-scopes is not responding correctly: it doesn't return repositories under the org.
2. the config-ui is sending `null` to the `PUT http://localhost:4000/api/plugins/github/connections/11/scopes` API

Am I right?

**Comment by petercort** [comment]
@klesh from what I can see that's correct. 

I've been running into a lot of issues setting up a debug environment to get more info about the calls, so I can't offer much more at the moment :(

**Comment by klesh** [comment]
There are a couple of bugs in the Github App implementation.
I suggest that you use tokens as a workaround for the time being, It might take quite a while to fix all those problems.

**Comment by sanjchhoker** [comment]
I had same issue , I did a workaround to add scopes to the connection programmatically using first github API to fetch repository id's and then add them to connection using devlake API.
However pipelines are generally flaky with the setup and have got a weird error on the 1st and last step of the pipeline without any actual error in the logs , with pipeline status set to Partially Successful.

**Comment by d4x1** [comment]
@sanjchhoker  Can you help submit a new issue and describe what happened in detail ?  It seems a differcent problem.

**Comment by petercort** [comment]
> Seems like there are multiple problems to me:
> 
> 1. http://localhost:4000/api/plugins/github/connections/11/remote-scopes is not responding correctly: it doesn't return repositories under the org.
> 2. the config-ui is sending `null` to the `PUT http://localhost:4000/api/plugins/github/connections/11/scopes` API
> 
> Am I right?

This behavior has been fixed in #8088 🎊
